### ISSUE TITLE
false positive with flow type and argument desctructuring

### ISSUE THREAD

**Issue by danez** [eslint/eslint#5021]
Assuming the following code snippet:
```js
type Props = { baseUrl: string };

export default ({ baseUrl, anchor, inspectUtilities }: Props) => {
  return null;
};
```

eslint reports 2 warnings:

```
  3:56  error  "Props" is already declared in the upper scope  no-shadow
  3:56  error  "Props" is defined but never used               no-unused-vars
```

There should be no warnings, as both babel and flow correctly work with this typeannotation.

---

eslint version: v1.10.3
plugins: react
os: mac os x 10.11

---

.eslintrc:
```json
{
  "extends": "airbnb",
  "parser": "babel-eslint",
  "env": {
    "node": true
  },
  "rules": {
    "react/prop-types": 0
  },
  "globals": {
    "ReactElement": false
  }
}
```

**Comment by eslintbot** [comment]
@danez Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I think you need https://github.com/zertosh/eslint-plugin-flow-vars to get this to work. ESLint doesn't support flow type annotations.

**Comment by danez** [comment]
Okay this plugin resolves the issue with the 'no-unused-vars' rule, but not with the 'no-shadow' rule.

**Comment by nzakas** [comment]
We don't natively support Flow. I'd suggest filing an issue with babel-eslint.
